port/common/omrsock.h

```diff
@@ -0,0 +1,87 @@
+/*******************************************************************************
+ * Copyright (c) 2019, 2019 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+#if !defined(OMRSOCK_H_)
+#define OMRSOCK_H_
+
+#include "omrportsock.h"
+
+/**
```

Comment:
I think you don't need to introduce data structures in this header. When you will add `port/unix/omrsock.c` and `port/win32/omrsock.c`, you won't be able to use `port/common/omrsock.h`. You will need to add `port/unix_include/omrsock.h` for `port/unix/omrsock.c` and `port/win32_include/omrsock.c` for `port/win32/omrsock.c`. @rwy0717 is this correct?

Impact: you will have a separate header for Windows and Unix platforms; you won't need to use ifdefs; code will be cleaner and easier to read.

Reply:
I see, that makes sense. I included `omrportsock.h` because `omrsock.c` then will only need to include `omrsock.h`. I could remove it, and then `omrsock.c` will just need to include both headers. Should I change it to the latter approach?

Should I add the unix/omrsock.h and win32/omrsock.h later on as we create the implementations?

Reply:
My comment was for the entire `common/omrsock.h `. It isn't focused on the inclusion of `omrportsock.h` in `omrsock.h`. At this point, you can remove `common/omrsock.h`, and directly include `omrportsock.h` in `common/omrsock.c`. My current understanding is that `common/omrsock.h` would only be consumed by `common/omrsock.c`. Since `common/omrsock.c` will only contain function stubs, you don't need to introduce platform specific data-structure definitions in `common/omrsock.h` [NOT needed]. 

For `[unix_include|win32_include]/omrsock.h`, you can include `omrportsock.h` in them.

> Should I add the unix_include/omrsock.h and win32_include/omrsock.h later on as we create the implementations?

Yes.

Reply:
Yes, I think you are correct. I see what you mean. Since only the data structures in `omrportsock.h` is needed by `omrsock.c`, we can safely remove `omrsock.h` for now. I tried building it on Linux and that works. I'll test with Jenkins on the other platforms.

Reply:
@babsingh I removed `common/omrsock.h` and ran the Jenkins test build for it and all platforms built successfully on JDK11. Ready for 2nd review.